main.sh

```diff
@@ -1,126 +1,534 @@
 #!/usr/bin/env bash
 
-cd $GITHUB_WORKSPACE
+#######################################
+# Echo error message.
+# Arguments:
+#   Message
+# Outputs:
+#   Writes ERROR: message to stdout.
+#######################################
+error_message() {
+  echo -en "\033[31mERROR\033[0m: $1"
+}
 
-COMMIT_ID=$(cat $GITHUB_EVENT_PATH | jq -r '.pull_request.head.sha')
+#######################################
+# Echo warning message.
+# Arguments:
+#   Message.
+# Outputs:
+#   Writes WARNING: message to stdout.
+#######################################
+warning_message() {
+  echo -en "\033[33mWARNING\033[0m: $1"
+}
+
+#######################################
+# Echo info message.
+# Arguments:
+#   Message.
+# Outputs:
+#   Writes INFO: message to stdout.
+#######################################
+info_message() {
+  echo -en "\033[32mINFO\033[0m: $1"
+}
 
-echo "COMMIT ID: $COMMIT_ID"
+if [[ "$GITHUB_EVENT_NAME" != "pull_request" ]]; then
+  echo $( warning_message "This action only runs on pull_request events." )
+  echo $( info_message "Refer https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request for more information." )
+  echo $( info_message "Exiting..." )
 
-PR_BODY=$(cat "$GITHUB_EVENT_PATH" | jq -r .pull_request.body)
-if [[ "$PR_BODY" == *"[do-not-scan]"* ]]; then
-  echo "[do-not-scan] found in PR description. Skipping PHPCS scan."
   exit 0
 fi
 
-stars=$(printf "%-30s" "*")
+if [[ $(cat "$GITHUB_EVENT_PATH" | jq -r .pull_request.body) == *"[do-not-scan]"* ]]; then
+  echo $( info_message "[do-not-scan] found in PR description. Skipping PHPCS scan." )
+
+  exit 0
+fi
 
-export RTBOT_WORKSPACE="/home/rtbot/github-workspace"
-hosts_file="$GITHUB_WORKSPACE/.github/hosts.yml"
+GITHUB_REPOSITORY_NAME=${GITHUB_REPOSITORY##*/}
+GITHUB_REPOSITORY_OWNER=${GITHUB_REPOSITORY%%/*}
+COMMIT_ID=$(cat $GITHUB_EVENT_PATH | jq -r '.pull_request.head.sha')
 
-# Delete all the folders to be skipped to ignore them from being scanned.
-if [[ -n "$SKIP_FOLDERS" ]]; then
+echo $( info_message "COMMIT_ID: $COMMIT_ID" )
+echo $( info_message "GITHUB_REPOSITORY_NAME: $GITHUB_REPOSITORY_NAME" )
+echo $( info_message "GITHUB_REPOSITORY_OWNER: $GITHUB_REPOSITORY_OWNER" )
 
-  folders=(${SKIP_FOLDERS//,/ })
+if [[ -z "$GITHUB_REPOSITORY_NAME" ]] || [[ -z "$GITHUB_REPOSITORY_OWNER" ]] || [[ -z "$COMMIT_ID" ]]; then
+  echo $( error_message "One or more of the following variables are not set: GITHUB_REPOSITORY_NAME, GITHUB_REPOSITORY_OWNER, COMMIT_ID" )
 
-  for folder in ${folders[@]}; do
-    path_of_folder="$GITHUB_WORKSPACE/$folder"
-    [[ -d "$path_of_folder" ]] && rm -rf $path_of_folder
-  done
+  exit 1
 fi
 
-rsync -a "$GITHUB_WORKSPACE/" "$RTBOT_WORKSPACE"
-rsync -a /root/vip-go-ci-tools/ /home/rtbot/vip-go-ci-tools
-chown -R rtbot:rtbot /home/rtbot/
+if [[ -n "$VAULT_TOKEN" ]]; then
+  GH_BOT_TOKEN=$(vault read -field=token secret/rtBot-token)
```

Comment:
`secret/rtBot` shouldn't be hardcoded here. @mrrobot47 thoughts?

Reply:
Add env variable for vault field and path. For backward compatibility, if the env is empty, default path to `secret/rtBot-token` and field to `token`

Reply:
I have bumped the vault to the latest version. Can you confirm if this command will still work?

Reply:
We are no longer using vault and plan on sunsetting it soon. So, for now, do not bump it. We will remove it in the coming releases.

Reply:
We are no longer using vault and plan on sunsetting it soon. We will remove it in the coming releases. I have not run the command to test it, but checked the documentation and this command is still valid.

Reply:
Should I add a deprecation annotation for this?

Reply:
Yes